File: src/main/java/models/Panier.java
package models;

public class Panier {
    private int id_panier;
    private String nomProduit;
    private double prix;
    private String etatStock;
    private int quantite;

    public Panier() {
    }
    public void ajouter(String nomProduit, double prix, String etatStock, int quantite) {
        this.nomProduit = nomProduit;
        this.prix = prix;
        this.etatStock = etatStock;
        this.quantite = quantite;
    }

    public Panier(int id_panier,String nomProduit, double prix, String etatStock, int quantite) {
        this.id_panier=id_panier;
        this.nomProduit = nomProduit;
        this.prix = prix;
        this.etatStock = etatStock;
        this.quantite = quantite;
    }

    public int getId() {
        return id_panier;
    }

    public void setId(int id_panier) {
        this.id_panier = id_panier;
    }

    public String getNomProduit() {
        return nomProduit;
    }

    public void setNomProduit(String nomProduit) {
        this.nomProduit = nomProduit;
    }

    public double getPrix() {
        return prix;
    }

    public void setPrix(double prix) {
        this.prix = prix;
    }

    public String getEtatStock() {
        return etatStock;
    }

    public void setEtatStock(String etatStock) {
        this.etatStock = etatStock;
    }

    public int getQuantite() {
        return quantite;
    }

    public void setQuantite(int quantite) {
        this.quantite = quantite;
    }

    @Override
    public String toString() {
        return "Panier{" +
                "id_panier=" + id_panier +
                ", nomProduit='" + nomProduit + '\'' +
                ", prix=" + prix +
                ", etatStock='" + etatStock + '\'' +
                ", quantite=" + quantite +
                '}';
    }
}

File: src/Entity/Panier.php
<?php

namespace App\Entity;

use App\Repository\PanierRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: PanierRepository::class)]
class Panier
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(type: 'integer')]
    private ?int $id;

    #[ORM\Column(type: 'string', length: 255)]
    private ?string $nomProduit;

    #[ORM\Column(type: 'decimal', precision: 10, scale: 2)]
    #[Assert\NotBlank]
    #[Assert\Type(type: 'numeric', message: 'The value {{ value }} is not a valid number.')]
    private ?float $prix; 

    #[ORM\Column(type: 'string', length: 255)]
    private ?string $etatStock;

    #[ORM\Column(type: 'integer')]
    private ?int $quantite;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomProduit(): ?string
    {
        return $this->nomProduit;
    }

    public function setNomProduit(string $nomProduit): self
    {
        $this->nomProduit = $nomProduit;

        return $this;
    }

    public function getPrix(): ?float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): self
    {
        $this->prix = $prix;

        return $this;
    }

    public function getEtatStock(): ?string
    {
        return $this->etatStock;
    }

    public function setEtatStock(string $etatStock): self
    {
        $this->etatStock = $etatStock;

        return $this;
    }

    public function getQuantite(): ?int
    {
        return $this->quantite;
    }

    public function setQuantite(int $quantite): self
    {
        $this->quantite = $quantite;

        return $this;
    }
}
